api/src/main/java/org/openmrs/module/reporting/cohort/definition/library/BuiltInCohortDefinitionLibrary.java

```diff
@@ -185,4 +188,31 @@ public CohortDefinition getPatientsWithEnrollment(){
             "locationList", "locations"
         );
     }
+
+    @DocumentedDefinition("patientsWithState")
+    public CohortDefinition getPatientsWithState() {
+        PatientStateCohortDefinition cd = new PatientStateCohortDefinition();
+        cd.addParameter(new Parameter("states", "reporting.parameter.states", ProgramWorkflowState.class, List.class, null));
+        cd.addParameter(new Parameter("startedOnOrAfter", "reporting.parameter.startDate", Date.class));
+        cd.addParameter(new Parameter("startedOnOrBefore", "reporting.parameter.endDate", Date.class));
+        cd.addParameter(new Parameter("endedOnOrAfter", "reporting.parameter.startDate", Date.class));
+        cd.addParameter(new Parameter("endedOnOrBefore", "reporting.parameter.endDate", Date.class));
+        return new MappedParametersCohortDefinition(cd,
+            "states", "states",
+            "startedOnOrAfter", "enrolledOnOrAfter",
+            "startedOnOrBefore", "enrolledOnOrBefore",
+            "endedOnOrAfter", "completedOnOrAfter",
+            "endedOnOrBefore", "completedOnOrBefore"
+        );
+    }
+
+    @DocumentedDefinition("patientsInState")
+    public CohortDefinition getPatientsInState() {
+        InStateCohortDefinition cd = new InStateCohortDefinition();
+        cd.addParameter(new Parameter("states", "reporting.parameter.states", ProgramWorkflowState.class, List.class, null));
+        cd.addParameter(new Parameter("onOrAfter", "reporting.parameter.startDate", Date.class));
+        cd.addParameter(new Parameter("onOrBefore", "reporting.parameter.endDate", Date.class));
+        cd.addParameter(new Parameter("onDate", "reporting.parameter.date", Date.class));
+        return cd;
```

Comment:
I'm not sure these labels are necessarily better than ones that reflect "onOrAfter", "onOrBefore", and "onDate".